### ISSUE TITLE
Option to delete / rename Case Metrics

### ISSUE THREAD

**Issue by crackytsi** [CERT-BDF/TheHive#225]
### Request Type
Feature-Request

### Work Environment

| Question              | Answer
|---------------------------|--------------------
| OS version (server)       | Debian 8
| TheHive version / git hash   | 2.11.1
| Package Type              | Debian package

### Problem Description
Currently Case Metrics can not be deleted or renamed.
Once you do a typo, it is not undo-able

**Comment by megan201296** [comment]
@crackytsi Just FYI, issue #72 is tracking deleting metrics and it is assigned to milestone 2.12.0. Figured I'd let you know it's on track :)

**Comment by nadouani** [comment]
Deleting metrics is not as simple as just drop them from metrics table.

The question is: how should we handle the metrics already used by existing cases. What would be the impact of deleting a metric...

Few questions to answer before going and add support to this feature

**Comment by crackytsi** [comment]
Yes I am aware of the complexity ;)

One (first) pragmatic idea:
-Rename means, keeping the metric, so just a log entry for the rename should be enough
-Make metric manually deletable from cases.
-If there are open cases using the metric ==> Metric is not deletable
-For closed cases add tag "metric-<key>-<value>-deleted

**Comment by megan201296** [comment]
@crackytsi has a good point of "If there are open cases using the metric ==> Metric is not deletable" ... At the minimum, we could delete metrics that are not being used by existing cases. That is good for accidentally created ones or ones we created but then decided not to use.

**Comment by crackytsi** [comment]
@nadouani Aren't case metrics very similar to case custom fields? Can't we apply the same way for deleting metrics as we use it for deleting custom fields?

**Comment by nadouani** [comment]
Yes, we can add the same for metrics. We didn't make in 3.4.0 for timing reasons.
